### DISCUSSION TITLE
How fast should I expect a fully cached command to take?

### DISCUSSION THREAD

**Discussion by lukehatcher** [vercel/turborepo#1101] [Help] [answered]
My question is, are there expectations for how fast a fully cached build/command should take? 

I have a rather large monorepo, (100k+ files) with 10 different workspaces. My "full turbo" with 10/10 cache hits takes me 6 seconds (~3mins un-cached). To test, I removed all but one workspace (they are all similar) and the "full turbo" 1/1 cache hit took ~2.5 seconds (~20s un-cached). Other smaller scripts in my monorepo all fail to get under the 2-3 second mark fully cached. 

Is this a sign that something is off? Obviously the speed improvements are still amazing but I want to make sure I am not leaving anything on the table.

Thanks for the help!

**Comment by mehulkar** [comment, marked as answer]
2.5s for 1 workspace does not seem right, but it's possible that when you deleted all those other workspaces you didn't delete a large package-lock.json or something and parsing that took time? A reproduction repo would be helpful here. You can also try running with `-vvv` to see if additional logs give you any insight about where time is being spent.
